test/CSharp/CodeCracker.Test/Refactoring/AllowMembersOrderingCodeFixProviderTests.Base.cs

```diff
@@ -0,0 +1,137 @@
+﻿using CodeCracker.Refactoring;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using TestHelper;
+using Xunit;
+
+namespace CodeCracker.Test.Refactoring
+{
+    public class BaseAllowMembersOrderingCodeFixProviderTests :
+        CodeFixTest<AllowMembersOrderingAnalyzer, BaseAllowMembersOrderingCodeFixProviderTests.MockCodeFixProvider>
+    {
+        [Theory]
+        [InlineData("class", "void B() { };", "void A() { };")]
+        [InlineData("struct", "int c = 3, d = 4;", "int a = 1, b = 2;")]
+        public async Task BaseAllowMembersOrderingShouldCallIComparerToOrder(string typeDeclaration, string memberA, string memberB)
+        {
+            var codeFixProvider = base.GetCSharpCodeFixProvider() as MockCodeFixProvider;
+
+            var source = @"
+            public " + typeDeclaration + @" Foo
+            {
+                " + memberA + @"
+                " + memberB + @" 
+            }";
+
+            var expected = @"
+            public " + typeDeclaration + @" Foo
+            {
+                " + memberB + @"
+                " + memberA + @" 
+            }";
+
+            await VerifyCSharpFixAsync(source, expected, codeFixProvider: codeFixProvider);
+            Assert.True(codeFixProvider.HasIComparerBeenCalled, "The IComparer must be used to sort the members of that type");
```

Comment:
I could use just VerifyCSharpAsync as a proof of IComparer was being called, but for readability sake I used a mock.